sys/dev/nmdm/nmdm.c

```diff
@@ -115,16 +115,16 @@ nmdm_close(struct tty *tp)
 		return;
 
 	/* Shut down self. */
+	tty_lock(tp);
 	tty_rel_gone(tp);
 
 	/* Shut down second part. */
-	tty_lock(tp);
 	onp = np->np_other;
 	if (onp == NULL)
 		return;
 	otp = onp->np_tty;
+	tty_lock(otp);
 	tty_rel_gone(otp);
-	tty_lock(tp);
```

Comment:
This needs to be restored because this tty needs to remain locked on exit, but we dropped it with `tty_rel_gone` again

Reply:
"Fixed" within the realm of the second and third patch.